aws/resource_aws_ram_resource_share_accepter.go

```diff
@@ -242,6 +262,12 @@ func resourceAwsRamResourceShareGetInvitation(conn *ram.RAM, resourceShareARN, s
 		return nil, fmt.Errorf("Error reading RAM resource share invitation %s: %s", resourceShareARN, err)
 	}
 
+	if *invitation.ReceiverAccountId != client.accountid {
+		// Report an error, this won't work on the long term since once the invitation has disappeared
+		// the receiver_account_id is populated with the account_id (and same during import)
+		return nil, fmt.Errorf("Unexpected behavior while reading RAM resource share invitation %s: receiver_account_id is %s while current account_id is %s", resourceShareARN, *invitation.ReceiverAccountId, client.accountid)
+	}
+
```

Comment:
I see no reason why this could happen, but I prefer to have the check in place to be notified if this happens under a corner-case condition, since this would cause the bugs fixed by this change to reappear under this corner-case condition.

Reply:
We may also want to remove this since its not guaranteed `client.accountid` is populated in the case it cannot be detected automatically via other means and the provider [`skip_requesting_account_id` argument](https://www.terraform.io/docs/providers/aws/index.html#skip_requesting_account_id) is set.